---
float64 next_x
float64 next_y